include/libtorrent/aux_/io.hpp

```diff
@@ -59,12 +59,16 @@ namespace libtorrent { namespace aux
 		return ret;
 	}
 
-	template <class T, class Byte>
+	template <class T, class In, class Byte, typename Cond
+		= typename std::enable_if<std::is_integral<T>::value &&
+		(std::is_integral<In>::value || std::is_enum<In>::value)>::type>
 	inline typename std::enable_if<sizeof(Byte)==1, void>::type
-	write_impl(T val, span<Byte>& view)
+	write_impl(In data, span<Byte>& view)
```

Comment:
I meant to say ``sizeof(Byte)`` must be 1. The code in here assumes that it has 8 bits specifically.